### PR TITLE
Convert `AlertListPopoverContent` to TS

### PR DESCRIPTION
**Pull request by oisincoveney**
Converts `AlertListPopoverContent` to TS. This PR doesn't modify any code except `AlertScheduleText`, which just changes the `span` to a Mantine `Text` component

No functionality should have changed, so all tests should pass.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertScheduleText.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+type AlertScheduleTextProps = {
+  schedule: Channel;
+  verbose?: boolean;
+};
+
+export const AlertScheduleText = ({
+  schedule,
+  verbose,
+}: AlertScheduleTextProps) => {
+  const getScheduleText = useCallback(() => {
```
Comment (heypoom) [inline frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertScheduleText.tsx:21]: imo, I would extract `getScheduleText({ schedule, verbose })` into its own function, so its possible to add tests to this function

also, I think we can do either ts-pattern or early returns here

**Review by heypoom — APPROVED** [review]
TS conversion looks good!

File: frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertScheduleText.tsx (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertScheduleText.tsx:21]: I'll add that in a separate PR so we can focus on that part a little more - I have some code for `ts-pattern` with this function but it's a bit of a mess with the extra null checks and things

File: frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+              >{jt`Unsubscribe`}</a>
+            )}
+            {!isAdmin && unsubscribingProgress && (
+              <span> {unsubscribingProgress}</span>
+            )}
+          </div>
+        </div>
+
+        {
+          // To-do: @kdoh wants to look into overall alignment
```
Comment (uladzimirdev) [inline frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertListItem.tsx:111]: wat? let's remove it

Comment (iethree) [reply frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertListItem.tsx]: 🤣  aww cmon, these are my favorite kinds of comments

**Comment by iethree** [comment]
FYI: 
- https://github.com/metabase/metabase/pull/46394

**Pull request merged**